apt-hook/hook.cc

```diff
@@ -240,29 +252,81 @@ static void process_template_file(
 }
 
 static void process_all_templates(
-   std::string esm_a_pkgs_count,
+   int esm_a_pkgs_count,
    std::string esm_a_pkgs,
-   std::string esm_i_pkgs_count,
+   int esm_i_pkgs_count,
    std::string esm_i_pkgs
 ) {
-   std::array<std::string, 2> template_file_names = {
-      CONTRACT_EXPIRY_STATUS_MESSAGE_TEMPLATE_PATH,
-      CONTRACT_EXPIRY_STATUS_APT_MESSAGE_TEMPLATE_PATH
+   std::array<std::string, 4> static_file_names = {
+      APT_PRE_INVOKE_APPS_PKGS_STATIC_PATH,
+      MOTD_APPS_PKGS_STATIC_PATH,
+      APT_PRE_INVOKE_INFRA_PKGS_STATIC_PATH,
+      MOTD_INFRA_PKGS_STATIC_PATH,
    };
-   std::array<std::string, 2> static_file_names = {
-      CONTRACT_EXPIRY_STATUS_MESSAGE_STATIC_PATH,
-      CONTRACT_EXPIRY_STATUS_APT_MESSAGE_STATIC_PATH
+   std::array<std::string, 3> apt_static_files = {
+      APT_PRE_INVOKE_APPS_PKGS_STATIC_PATH,
+      APT_PRE_INVOKE_INFRA_PKGS_STATIC_PATH,
+      UBUNTU_NO_WARRANTY_STATIC_PATH
    };
+   std::array<std::string, 3> motd_static_files = {
+      MOTD_APPS_PKGS_STATIC_PATH,
+      MOTD_INFRA_PKGS_STATIC_PATH,
+      UBUNTU_NO_WARRANTY_STATIC_PATH
+   };
+
+   std::vector<std::string> template_file_names;
+   if (esm_a_pkgs_count > 0) {
+      template_file_names.push_back(APT_PRE_INVOKE_APPS_PKGS_TEMPLATE_PATH);
+      template_file_names.push_back(MOTD_APPS_PKGS_TEMPLATE_PATH);
+   } else {
+      template_file_names.push_back(APT_PRE_INVOKE_APPS_NO_PKGS_TEMPLATE_PATH);
+      template_file_names.push_back(MOTD_APPS_NO_PKGS_TEMPLATE_PATH);
+   }
+   if (esm_i_pkgs_count > 0) {
+      template_file_names.push_back(APT_PRE_INVOKE_INFRA_PKGS_TEMPLATE_PATH);
+      template_file_names.push_back(MOTD_INFRA_PKGS_TEMPLATE_PATH);
+   } else {
+      template_file_names.push_back(APT_PRE_INVOKE_INFRA_NO_PKGS_TEMPLATE_PATH);
+      template_file_names.push_back(MOTD_INFRA_NO_PKGS_TEMPLATE_PATH);
+   }
    for (uint i = 0; i < template_file_names.size(); i++) {
       process_template_file(
          template_file_names[i], 
          static_file_names[i], 
-         esm_a_pkgs_count,
+         std::to_string(esm_a_pkgs_count),
          esm_a_pkgs,
-         esm_i_pkgs_count,
+         std::to_string(esm_i_pkgs_count),
          esm_i_pkgs
       );
    }
+
+   std::ofstream apt_pre_invoke_msg;
+   apt_pre_invoke_msg.open(APT_PRE_INVOKE_MESSAGE_STATIC_PATH);
+   for (uint i = 0; i < apt_static_files.size(); i++) {
+       std::ifstream message_file(apt_static_files[i]);
+       apt_pre_invoke_msg << std::endl;
+       if (message_file.is_open()) {
+           apt_pre_invoke_msg << message_file.rdbuf();
+           message_file.close();
+       };
```

Comment:
I think we need to move this line
```suggestion
       if (message_file.is_open()) {
           apt_pre_invoke_msg << std::endl;
           apt_pre_invoke_msg << message_file.rdbuf();
           message_file.close();
       };
```

to avoid adding a new line for each file unless the file exists

Reply:
excellent.thanks